server/src/main/java/org/elasticsearch/common/lucene/search/function/ScriptScoreQuery.java

```diff
@@ -143,7 +148,12 @@ public Explanation explain(LeafReaderContext context, int doc) throws IOExceptio
                         explanation = Explanation.match(score, desc);
                     }
                 }
-                
+                if (boost != 1.0f) {
+                    List<Explanation> subs = new ArrayList<>();
+                    subs.addAll(Arrays.asList(explanation.getDetails()));
+                    subs.add(Explanation.match(boost, "boost"));
+                    explanation = Explanation.match(explanation.getValue(), explanation.getDescription(), subs);
+                }
```

Comment:
I'd suggest wrapping the explanation instead of modifying it in-place: create the scorer with boost=1f a couple lines above, and then here:

```java
if (boost != 1f) {
    explanation = Explanation.match(boost * explanation.getValue().floatValue(), "Boosted score, product of:",
        Explanation.match(boost, "boost"),
        explanation);
}
```